Sources/Bitcoin/BitcoinAddress.swift

```diff
@@ -0,0 +1,83 @@
+// Copyright © 2017-2018 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+import Foundation
+
+public struct BitcoinAddress: Address, Hashable {
```

Comment:
Since both `BitcoinAddress` and `EthereumAddress` conform to `Hashable`, should we make `Address` conform to `Hashable` then?

Reply:
No because that will make `Address` a PAT.

Reply:
what is PAT?

Reply:
https://www.natashatherobot.com/swift-what-are-protocols-with-associated-types/

Reply:
more relevant link: https://appventure.me/2017/12/10/patterns-for-working-with-associated-types/